### PR TITLE
Added function to transfer funds from/to spot account and futures account

### PR DESCRIPTION
**Pull request by chrisghill**
I noticed that your library only implemented the "get" HTTP function on the transfer endpoint, which queries for transfer history.  I added the "post" version which allows you to transfer funds to/from your futures account from your spot account.

I noticed for margin trading, you explicitly made separate functions for each transfer direction: https://github.com/sammchardy/python-binance/blob/master/binance/client.py#L2500-L2558

But it seems the standard account stuff was more fleshed out in terms of documentation and function parameters.  The futures stuff seems to be left to the developer to read API docs, thus I did it that way.  I can implement separate to/from functions if you prefer.  Just let me know.

### CODE REVIEW

**Comment by kimchirichie** [comment]
i've been looking for this

File: binance/client.py [hunk 1]
```diff
@@ -3567,10 +3567,18 @@ def futures_leverage_bracket(self, **params):
...
+        https://binance-docs.github.io/apidocs/futures/en/#new-future-account-transfer
+
+        """
+        return self._request_margin_api('post', 'futures/transfer', True, data=params)
+
     def transfer_history(self, **params):
         """Get future account transaction history list
 
-        https://binance-docs.github.io/apidocs/futures/en/#new-future-account-transfer
+        https://binance-docs.github.io/apidocs/futures/en/#get-future-account-transaction-history-list-user_data
```
Comment (kimchirichie) [inline binance/client.py:3581]: harmless but unrelated to PR

Comment (chrisghill) [reply binance/client.py:3581]: Actually, I think it is related to the PR (if I remember correctly).  That comment links the POST endpoint documentation, but is surrounding the code for the GET endpoint (2 different actions).   I implemented the POST endpoint, and linked POST endpoint documentation and this specific line is me updating the GET function call to link the GET endpoint documentation.

Comment (kimchirichie) [reply binance/client.py:3581]: my bad, that makes sense

Comment (chrisghill) [reply binance/client.py:3581]: No worries. Thanks for reviewing! :)

**Comment by kimchirichie** [comment]
if we can get someone to verify this works as expected I can merge this in

**Comment by chrisghill** [comment]
@kimchirichie I personally have been using this capability for a while in my own application and can confirm it works, but if you need somebody else to confirm that's understandable.

File: binance/client.py (hunk 1 above)
Comment (avatar-lavventura) [reply binance/client.py:3581]: @chrisghill How do you call the function? I have tried `client.futures_account_transfer(asset="USDT", amount="1")` but keep getting following error message: `Mandatory parameter 'type' was not sent, was empty/null, or malformed.`. What `type` argument should we provide?

**Pull request merged**